<?xml version='1.0'?>

<launch>
    <!-- Input Arguments -->
    <arg name="confidence_threshold" default="0.5" />
    <arg name="iou_threshold" default="0.6" />
    <arg name="model_path" default="/src/yolov5" />
    <arg name="weights_path" default="/model/bestv8nv2.pt" />
    <arg name="classes_path" default="/perception_pkg/model/data.yaml" />

    <!-- Input Arguments -->
    <arg name="image_rgb_topic_name" default="/image" />
    <arg name="image_depth_topic_name" default="/image_depth" />
    <arg name="odom_topic_name" default="/odom" />
    <arg name="prespective_angle_x" default="60" />    
    <arg name="resolution_x" default="960" />
    <arg name="resolution_y" default="480" />
    
    <node pkg="perception_pkg"  name="object_detection" type="object_detection_v8.py" output="screen">
        <param name="confidence_threshold" value="$(arg confidence_threshold)" />
        <param name="iou_threshold" value="$(arg iou_threshold)" />
        <param name="model_path" value="$(arg model_path)" />
        <param name="weights_path" value="$(arg weights_path)" />
        <param name="classes_path"  value="$(arg classes_path)" />
    </node>

    <!-- Launch Lane Detector Node -->
    <node pkg="lane_detection_canny" type="lane_detection.py" name="lane_detection" output="screen">
        <param name="image_rgb_topic_name" value="$(arg image_rgb_topic_name)" />
        <param name="image_depth_topic_name" value="$(arg image_depth_topic_name)" />
        <param name="odom_topic_name" value="$(arg odom_topic_name)" />
        <param name="prespective_angle_x" value="$(arg prespective_angle_x)" />
        <param name="resolution_x" value="$(arg resolution_x)" />
        <param name="resolution_y" value="$(arg resolution_y)" />
    </node>


</launch>
